### PR TITLE
feat : validate time schedules to be one hour apart

### PR DESCRIPTION
**Pull request by actlikewill**
#### Description
- Adds the validation for time schedules. They need to be one hour apart. 

#### Type of change
<pre>
+ Added something.
~ Modified something.
</pre>

### CODE REVIEW

File: src/utils/validateTimeOneHour.ts [hunk 1]
```diff
@@ -1,21 +1,59 @@
...
-const validateOneHourTime = (schedules: string[], newTime: string): boolean => {
-	let diff: number | undefined = 0;
-	const oneHour = 60 * 60 * 1000;
+export const validateNewOneHourTime = (
+	schedules: string[],
+	newTime: string,
+): boolean => {
+	if (schedules.length === 0) return true;
+	const diff = getDiff(newTime, schedules[schedules.length - 1]);
+	return diff < -3600000;
```
Comment (mashafrancis) [inline src/utils/validateTimeOneHour.ts:13]: I think for it to hit the one hour block, we can have the difference to be < 3599999. The one second is being validated as well. For instance, if I have a schedule at 3:00pm, the next possible one would be at 4:01pm, and the next at 5:02pm. We can adjust the number of seconds to be 1 less to cater for this. At least and exact interval at 3:00, 4:00 etc would follow. What do you think about it?

File: src/utils/validateTimeOneHour.ts [hunk 2]
```diff
@@ -1,21 +1,59 @@
...
+export const validateEditOneHourTime = (
+	schedules: any,
+	scheduleId: string,
+	editTime: string,
+): boolean => {
+	const editScheduleIndex = schedules.findIndex(
+		(item) => item._id === scheduleId,
+	);
+	const timeBefore =
+		editScheduleIndex === 0 ? null : schedules[editScheduleIndex - 1].schedule;
```
Comment (mashafrancis) [inline src/utils/validateTimeOneHour.ts:29]: You can add a check for null or undefined values. At one 22, you can have it as `...schedules[editScheduleIndex - 1]?.schedule` with the question operator.

**Review by mashafrancis — COMMENTED** [review]
Nice work @actlikewill. This works as expected.
I've added a few enhancements too. You can check them out as well.

File: src/utils/validateTimeOneHour.ts (hunk 1 above)
Comment (actlikewill) [reply src/utils/validateTimeOneHour.ts]: I have adjusted the comparison to be <= 3600000 which is what I actually had in the first place. I think I changed it while I was testing. It should work now with exact one hour values. You can confirm.

File: src/utils/validateTimeOneHour.ts (hunk 2 above)
Comment (actlikewill) [reply src/utils/validateTimeOneHour.ts:29]: I fixed the error that was causing that bug. We acutally don't need to validate the initial value if we are editing a schedule. I hope that now works okay. Please confirm.

File: src/utils/validateTimeOneHour.ts (hunk 1 above)
Comment (mashafrancis) [reply src/utils/validateTimeOneHour.ts]: Great. Much better

**Comment by mashafrancis** [comment]
Awesome work. LGTM.

**Pull request merged**